docs(contributor-guide): update v0.7 indexing (#840)
GreptimeTeam/docs: docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md

![Puffin](/puffin.png)

Currently, the inverted index serves as the first separate index structure supported, stored in the index file in the form of a Blob.

## Inverted Index

In version 0.7, GreptimeDB introduced the inverted index to accelerate queries.

The inverted index is a common index structure used for full-text searches, mapping each word in the document to a list of documents containing that word. We have applied this technology, originating from search engines, to the time series database GreptimeDB.

Search engines and time series databases belong to two different domains. Although the principle of the inverted index technology applied is similar, there are some conceptual migrations:
1. Term: In GreptimeDB, it refers to the column value of the time series.
2. Document: In GreptimeDB, it refers to the data segment containing multiple time series.

The introduction of the inverted index allows GreptimeDB to skip data segments that do not meet the conditions during queries, thereby improving scanning performance.

![Inverted index searching](/inverted-index-searching.png)

> For example, in the above query, retrieve the data segments where `job` = `apiserver`, `handler` matches the regex `.*users`, and `status` matches the regex `4..` from the inverted index, and get the data segment list `[1]`. Subsequently, only the data segment needs to be scanned, reducing a large number of IO operations.

Reviewer: ```suggestion
For instance, the query above uses the inverted index to identify data segments where `job` equals `apiserver`, `handler` matches the regex `.*users`, and `status` matches the regex `4...`. It then scans these data segments to produce the final results that meet all conditions, significantly reducing the number of IO operations.
```

### Inverted Index Format

> The inverted index is stored by column internally, and each column's inverted index block consists of FST and a Bitmap storing data segment IDs.

Reviewer: This description confuses user, but I don't know how to modify it...

Author: I rephrased it, please take a look.

> The FST (Finite State Transducer) allows us to store mappings from column values to Bitmap positions in a compact format and provides excellent search performance and complex search capabilities (such as regular expression matching).

Reviewer: ```suggestion
The FST (Finite State Transducer) allows GreptimeDB to store mappings from column values to Bitmap positions in a compact format and provides excellent search performance and supports complex search capabilities (such as regular expression matching).
```

> The Bitmap is used to store data segment IDs, with each bit representing a data segment, and a 1 indicates that the data segment contains the column value.

Reviewer: It's also confusing here, where does `1` come from? what is `the column value`?

![Inverted index format](/inverted-index-format.png)

### Index Data Segments

> GreptimeDB divides an SST file into several indexed data segments, each containing an equal number of rows.

Reviewer: ```suggestion
GreptimeDB divides an SST file into multiple indexed data segments, with each segment housing an equal number of rows.
This segmentation is designed to optimize query performance by scanning only the data segments that match the query conditions. 
```

> For example, if a data segment contains 1024 rows and the data segment list obtained after applying the inverted index to the query conditions is `[0, 2]`, then only the 0th and 2nd data segments in the SST file need to be scanned, i.e., from row 0 to row 1023 and from row 2048 to row 3071.

Reviewer: ```suggestion
For example, if a data segment contains 1024 rows and the list of data segments identified through the inverted index for the query conditions is `[0, 2]`, then only the 0th and 2nd data segments in the SST file—from rows 0 to 1023 and 2048 to 3071, respectively—need to be scanned.
```
